lib/pbench/test/unit/server/query_apis/conftest.py

```diff
@@ -35,20 +34,38 @@ def query_api(client, server_config, requests_mock, request):
     request_mock post method. The most common are 'json' for the JSON
     response payload, and 'exc' to throw an exception.
 
+    NOTE: the expected URI for the server query and the resulting Elasticsearch
+    query come from a test case parameter, which is passed through the Pytest
+    "mark" mechanism and accessed through the "param" attribute of the Pytest
+    request fixture. E.g.,
+
+            @pytest.mark.parametrize(
+                "query_api",
+                [
+                    {
+                        "pbench": "/controllers/list",
+                        "elastic": "/_search?ignore_unavailable=true",
+                    }
+                ],
+                indirect=True,
+            )
+
+    The first parameter must match the name of this fixture, and the second should
+    be a list (or tuple) with a single dict defining the "pbench" and "elastic"
+    URI paths.
+
     :return: the response object for further checking
     """
 
     def query_api(payload, expected_index, expected_status, server_config, **kwargs):
         host = server_config.get("elasticsearch", "host")
         port = server_config.get("elasticsearch", "port")
-        es_url = f"http://{host}:{port}"
-        requests_mock.post(re.compile(f"{es_url}"), **kwargs)
+        es_url = f"http://{host}:{port}" + expected_index + request.param["elastic"]
```

Comment:
The mixture of f-string and concatenation is slightly odd...I think I would recommend one or the other.  E.g., I think
```
        es_url = f"http://{host}:{port}{expected_index}{request.param['elastic']}"
```
is more compact.

Reply:
Yeah; a result of fixing it by cutting and pasting and not paying enough attention.